### ISSUE TITLE
Bug: Removes <!DOCTYPE>

### ISSUE THREAD

**Issue by Empereol** [stringham/angular-template-formatter#4]
## Expected Behavior
Doesn't remove `<!DOCTYPE html>` after format

## Actual Behavior
Removes `<!DOCTYPE html>` after format

## Steps to Reproduce the Problem

  1.  Format a document that contains `<!DOCTYPE>`

---

I was poking around and it appears that the result after the `src` is passed into `htmlParser.parse()` doesn't return a `<!DOCTYPE>` but instead the `rootNodes` first item is just `{value: ""}`.

This wouldn't be so much of an issue if you could disable this formatter for non-Angular HTML documents... But I'm not sure of an easy way to achieve this since, AFAIK, the Angular HTML templates aren't explicitly their own "language".

We need something like this:

```json
...
"activationEvents": [
    "onLanguage:angular-html"
],
...
```

But of course that would mean VS Code understanding what `angular-html` is... Anyway to achieve that?

**Comment by janKir** [comment]
I have also experienced exactly this issue just now.

And before, I also had some trouble with other "normal" tags being stripped away. I don't know if there was some syntax error in my html when this happened, but I found it extremely hard to identify the source of error and to recreate the stripped html. In general, for me this was unexpected behavior that a formatter extension is actually deleting code. I would expect this extension to only add/remove whitespace. I think a better way a better way of handling syntax errors would be to output some kind of error instead of just deleting code.

By the way: I still love this extension, good work!
